### ISSUE TITLE
Add Non-linear Cart-Pole model

### ISSUE THREAD

**Issue by mforets** [JuliaReach/NeuralNetworkAnalysis.jl#1]
- https://easychair.org/publications/open/BFKs

**Comment by SebastianGuadalupe** [comment]
WIP model:
```julia
@taylorize function cartpole!(du, u, p, t)
    local f, m, l, mt, g = 10, 0.1, 0.5, 1.1, 9.8
    sinθ = sin(u[3])
    cosθ = cos(u[3])
    aux = (f + m*l*u[4]^2*sinθ) / mt
    aux2 = (g*sinθ- cosθ) * aux / l*(4/3 - m*cosθ^2/mt)
    aux3 = cosθ/mt
    
    du[1] = u[2]
    du[2] = aux - m*l*aux2 * aux3
    du[3] = u[4]
    du[4] = aux2 / l*(4/3 - m*cosθ^2/mt) * aux3
    
    return du
end

# define the initial-value problem
X₀ = Hyperrectangle(low=[-0.05, -0.05, -0.05, -0.05], high=[0.05, 0.05, 0.05, 0.05])

prob = @ivp(x' = cartpole!(x), dim: 4, x(0) ∈ X₀)

# solve it
sol = solve(prob, T=1.0, alg=TMJets(max_steps=20_000, abs_tol=1e-10));
```

**Comment by mforets** [comment]
looks good! seems like we can also refactor the term `m*cosθ^2/mt`.

**Comment by SebastianGuadalupe** [comment]
Fixed by #8.
